frontend/packages/console-dynamic-plugin-sdk/src/extensions/resource-metadata.ts

```diff
@@ -13,14 +13,18 @@ export type ModelMetadata = ExtensionDeclaration<
     color?: string;
     /** Override the label. Requires `kind` be provided. */
     label?: string;
+    /** Override plural. Requires `kind` be provided. */
+    plural?: string;
     /** Override the plural label. Requires `kind` be provided. */
     labelPlural?: string;
     /** Customize the abbreviation. Defaults to All uppercase chars in the kind up to 4 characters long. Requires `kind` be provided. */
     abbr?: string;
+    /** Flag if the model is a Custom Resource Definition */
+    crd?: boolean;
```

Comment:
Why would `crd` be part of the metadata?
It should be known at discovery time whether the model is a CRD or not and therefore it isn't required to be augmented by an extension.

Reply:
`crd` is a misnomer. IIRC, it just determines whether we use the fully-qualified kind/group/version in the URL instead of the short plural name (for legacy paths we don't want to break). IMO all models contributed by plugins should have `crd: true`, and there shouldn't be a way to override.

Reply:
Ok so i think we can agree that we can omit this value in the extension and internally set it to `true` if needed.